src/NServiceBus.Gateway.AcceptanceTests/When_sending_fails_with_retries.cs

```diff
@@ -0,0 +1,170 @@
+            {
+                c.Id = Guid.NewGuid();
+            })
+            .WithEndpoint<HeadquartersWithCustomRetryPolicy>(b => b.When(async (bus, context) => await bus.SendToSites(new[] { "SiteA" }, new AnyMessage { Id = context.Id })))
+            .WithEndpoint<ErrorSpy>()
+            .Done(c => c.MessageMovedToErrorQueue)
+            .Repeat(r => r.For(Transports.Default))
+            .Should(c => Assert.True(c.CustomRetryPolicyWasCalled,"Custom retry policy was not called"))
+            .Run();
+        }
+
+        class Context : ScenarioContext
+        {
+            public Guid Id { get; set; }
+            public bool MessageMovedToErrorQueue { get; set; }
+            public int NumberOfRetries { get; set; }
+            public bool CustomRetryPolicyWasCalled { get; set; }
+        }
+
+        class HeadquartersWithDefaultRetryPolicy : EndpointConfigurationBuilder
+        {
+            public HeadquartersWithDefaultRetryPolicy()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.ScaleOut().InstanceDiscriminator("1");
+                    c.EnableFeature<Gateway>();
+                    c.EnableFeature<TimeoutManager>();
+                    c.SendFailedMessagesTo(ErrorSpyQueueName);
+                    c.Gateway().Retries(3,TimeSpan.FromSeconds(1));
+                })
+                    .WithConfig<GatewayConfig>(c =>
+                        {
+                            c.Sites = new SiteCollection
+                                {
+                                    new SiteConfig
+                                        {
+                                            Key = "SiteA",
+                                            Address = "http://localhost:25999/SiteA/",
+                                            ChannelType = "http"
+                                        }
+                                };
+
+                            c.Channels = new ChannelCollection
+                                {
+                                    new ChannelConfig
+                                        {
+                                             Address = "http://localhost:25999/Headquarters/",
+                                            ChannelType = "http"
+                                        }
+                                };
+                        });
+            }
+        }
+
+        class HeadquartersWithCustomRetryPolicy : EndpointConfigurationBuilder
+        {
+            public HeadquartersWithCustomRetryPolicy()
+            {
+                EndpointSetup<DefaultServer>((c, d) =>
+                {
+                    c.ScaleOut().InstanceDiscriminator("1");
+                    c.EnableFeature<Gateway>();
+                    c.EnableFeature<TimeoutManager>();
+                    c.SendFailedMessagesTo(ErrorSpyQueueName);
+                    c.Gateway().CustomRetryPolicy((msg, ex, currentRetry) =>
```

Comment:
is this the only difference to `HeadquartersWithDefaultRetryPolicy`? In that case you can set that up on ` .WithEndpoint` within the testcase itself.